File: lucette2/src/test/scala/battle/csn/lucette2/board/DameBoardTest.scala
package battle.csn.lucette2.board

import org.scalatest.junit.AssertionsForJUnit
import org.junit.runner.RunWith
import org.mockito.runners.MockitoJUnitRunner
import battle.csn.lucette2.game.logic.NegaMax
import org.junit.Assert._
import org.junit._
import org.scalatest.mock._
import org.mockito.Mockito._
import battle.csn.lucette2.game.board.DameBoard

@RunWith(classOf[MockitoJUnitRunner])
class DameBoardTest {

  @Test
  def moveAvailablesTest() {
    val board = new DameBoard()
    println("search for : ")
    println(board)
    val ms = board.moveAvailables(1)
    println("find : "+ms.size)
    ms.foreach(println(_))
  }

  @Test
  def toStringTest() {
    val board = new DameBoard()
    val res = board.toString()
    board.updateBoard(res)
    val res2 = board.toString
    assertTrue(res.equals(res2))
  }

}

File: trunk/lucette2/src/test/java/battle/csn/lucette/board/DameBoardTest.java
package battle.csn.lucette.board;

import static org.junit.Assert.assertTrue;

import java.util.List;

import org.junit.Before;
import org.junit.Test;
import org.junit.runner.RunWith;
import org.junit.runners.JUnit4;

import battle.csn.lucette.game.board.DameBoard;
import battle.csn.lucette.game.structure.Move;


@RunWith( JUnit4.class )
public class DameBoardTest
{
    DameBoard board;

    @Before
    public void initBoard( )
    {
        board = new DameBoard( );
    }

    @Test
    public void compareWithScala( )
    {
        List<Move> moves = board.getMoveAvailables( );
        System.out.println( board );
        System.out.println( moves.size( ) );
        for ( Move m : moves )
        {
            System.out.println( m );
        }
    }

    @Test
    public void getBoardTest( )
    {
        String strBoard1 = board.getBoard( );
        assertTrue( strBoard1.charAt( 23 ) == 'x' );
        board.play( new Move( 1, 1, 2, 0 ) );
        String strBoard2 = board.getBoard( );
        assertTrue( strBoard2.charAt( 23 ) == ' ' );
    }

    @Test
    public void updateBoardTest( )
    {
        for ( Move move : board.getMoveAvailables( ) )
        {
            DameBoard b1 = new DameBoard( );
            DameBoard b2 = new DameBoard( );
            b1.play( move );
            String strBoard1 = b1.getBoard( );
            b2.updateBoard( strBoard1 );
            String strBoard2 = b2.getBoard( );
            assertTrue( strBoard1.equals( strBoard2 ) );
        }
    }
}
